content/docs/introduction/faq.md

```diff
@@ -153,3 +163,49 @@ should take less than a minute under normal circumstances. See [crash recovery](
 
 You have run into a bug of ZFS on Linux. See [issue #484](https://github.com/prometheus/prometheus/issues/484)
 for details. Upgrading to ZFS on Linux v0.6.4 should fix the issue.
+
+## Implementation
+
+### Why are all sample values 64-bit floats? I want integers.
+
+We restrained ourselves to 64-bit floats to simplify the design. The
+[IEEE 754 double-precision binary floating-point
+format](http://en.wikipedia.org/wiki/Double-precision_floating-point_format)
+supports integer precision for values up to 2<sup>53</sup>. Supporting
+native 64 bit integers would (only) help if you need integer precision
+above 2<sup>53</sup> but below 2<sup>63</sup>. In principle, support
+for different sample value types (including some kind of big integer,
+supporting even more than 64 bit) could be implemented, but it is not
+a priority right now. Note that a counter, even if incremented
+one million times per second, will only run into precision issues
+after more than 285 years.
+
+### Why does Prometheus use a custom storage backend rather than [some other storage system]? Isn't the "one file per time series" approach killing performance?
+
+Initially, Prometheus ran completely on LevelDB, but to achieve better
+performance, we had to change the storage for bulk sample data. We
+evaluated many storage systems that were available at the time,
+without getting satisfactory results. So we implemented exactly the
+parts we needed, while keeping LevelDB for indexes and making heavy
+use of file system capabilities. Obviously, we could not evaluate
+every single storage system out there, and storage systems have
+evolved meanwhile. However, the performance of the solution
+implemented now is satisfactory for most use-cases. After an
+acceptable query speed for common queries, an important metric is the
+sustainable ingestion rate. It depends on the compressibility of the
+sample data and on the number of time series the samples belong to,
+but to give you an idea, here are some results from benchmarks:
+
+* On an older 8-core machine with Intel Core i7 CPUs and two spinning
+  disks (Samsung HD753LJ) in a RAID-1 setup, Prometheus could ingest
+  50k samples per seconds, belonging to 300k time series. (TODO:
+  Reproduce values. Identify bottleneck.)
+* On a modern server with SSD, Prometheus could ingest more than 100k
+  samples per second, belonging to millions of time series. The
+  bottleneck was identified as the hash calculation, which happens
```

Comment:
I'm not sure if that strictly applies. The hash calculation is happening sequentially in a pipeline. It could be the overhead of the channel receive and send, and the fact that it is single-threaded at that point...